src/commands/games/blackjack.ts

```diff
@@ -0,0 +1,247 @@
+    gameMessage: Message,
+    playerId: string
+  ): Promise<GameState | null> {
+    // Collect the first reaction
+    const reaction = collected.first();
+    // Remove the user's reaction once we've received it
+    if (reaction) await gameMessage.reactions.resolve(reaction)?.users.remove(playerId);
+    // Perform action according to reaction
+    switch (reaction?.emoji.name) {
+      case '🇸':
+        return performGameAction(playerId, BlackjackAction.STAND);
+      case '🇭':
+        return performGameAction(playerId, BlackjackAction.HIT);
+      case '🇶':
+        return performGameAction(playerId, BlackjackAction.QUIT);
+      default:
+        return null;
+    }
+  }
+
+  /*
+    Waits for player's reaction and handle any corresponding actions.
+  */
+  private async handlePlayerAction(gameMessage: Message, playerId: string) {
+    const reactFilter = (reaction: MessageReaction, user: User) => this.reactFilter(reaction, user, playerId);
+    // only waits for 1 valid reaction from the player, with a time limit of 1 minute.
+    const reactCollector = await gameMessage.awaitReactions(reactFilter, { max: 1, time: 60000, errors: ['time'] });
+    // perform action corresponding to reaction
+    return await this.performActionFromReaction(reactCollector, gameMessage, playerId);
+  }
+
+  /*
+    Returns a colour string depending on the game's state
+  */
+  private getEmbedColourFromGame(game: GameState): string {
+    if (game.stage === BlackjackStage.DONE) {
+      if (this.getBalanceChange(game) < 0) {
+        // player lost coins
+        return 'RED';
+      }
+      if (this.getBalanceChange(game) > 0) {
+        // player won coins
+        return 'GREEN';
+      }
+      // player didn't lose any coins
+      return 'ORANGE';
+    }
+    // game in progress
+    return 'YELLOW';
+  }
+
+  /*
+    Returns the amount the player gains or loses from the game's current state.
+  */
+  private getBalanceChange(game: GameState): number {
+    return game.amountWon - game.bet;
+  }
+
+  /*
+    Returns a description of the game given the game's current state
+  */
+  private getDescriptionFromGame(game: GameState): string {
+    const amountDiff = Math.abs(this.getBalanceChange(game));
+    if (game.stage === BlackjackStage.DONE) {
+      if (game.surrendered) {
+        // player surrendered
+        return `You surrendered and lost **${amountDiff}** Codey coin(s) ${getEmojiByName('codeySad')}.`;
+      }
+      if (game.amountWon < game.bet) {
+        // player lost
+        return `You lost **${amountDiff}** Codey coin(s) ${getEmojiByName('codeySad')}, better luck next time!`;
+      }
+      if (game.amountWon > game.bet) {
+        // player won
+        return `You won **${amountDiff}** Codey coin(s) ${getEmojiByName('codeyLove')}, keep your win streak going!`;
+      }
+      // player tied with dealer
+      return `Tied! You didn't win or lose any Codey coin(s), try again!`;
```

Comment:
"or" -> "nor" probably.

Reply:
oops i set it at 100k before but changed to 1m good catch

Reply:
yeah I think that edge case is fine, I hope none of the admins remove those embeds, even if the bot crashes, the runtime will restart the bot immediately so not a big deal